include/grpc/grpc_crl_provider.h

```diff
@@ -0,0 +1,79 @@
+//
+//
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#ifndef GRPC_GRPC_CRL_PROVIDER_H
+#define GRPC_GRPC_CRL_PROVIDER_H
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+
+#include <grpc/support/sync.h>
+
+namespace grpc_core {
+namespace experimental {
+
+// Opaque representation of a CRL
+class Crl {
+ public:
+  static absl::StatusOr<std::unique_ptr<Crl>> Parse(
```

Comment:
The intention is that users can use this function to parse into our `Crl` type if they are implementing their own provider - a `CrlProvider` has to return `std::shared_ptr<Crl>` in the `GetCrl` implementation

Reply:
My concern is just that `Parse` uses our default `Crl` implementation, which is not clear from the current code (at least without looking at the implementation). One way to make this would be to move it out of the `Crl` class, but we could also just document what `Parse` does in a comment.

Reply:
> My concern is just that Parse uses our default Crl implementation

I'm still not quite clear - there's only 1 CRL implementation. `Crl` is our opaque API representation, and internally it will be backed by the concrete `CrlImpl`. The user won't be implementing their own `Crl` type